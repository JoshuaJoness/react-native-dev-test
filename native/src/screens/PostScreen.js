import React from 'react'
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native'
import Markdown from 'react-native-markdown-display';

const PostScreen = ({ navigation }) => {
  const title = navigation.getParam('title')
  const body = navigation.getParam('body')

  return (
    <>
      <Text>{title}</Text>
      <Markdown>{body}</Markdown>
    </>
  )
}

const styles = StyleSheet.create({})

export default PostScreen
