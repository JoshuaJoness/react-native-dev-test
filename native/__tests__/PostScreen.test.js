import React from 'react';
import PostScreen from '../src/screens/PostScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  jest.useFakeTimers();
  const tree = renderer.create(<PostScreen navigation={{ getParam: jest.fn() }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
