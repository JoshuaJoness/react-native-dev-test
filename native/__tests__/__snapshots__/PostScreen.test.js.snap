// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Object {
      "flex": 1,
      "marginBottom": 40,
      "marginHorizontal": 20,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 40,
          "fontWeight": "bold",
          "marginVertical": 15,
        }
      }
    />
    <View
      style={Object {}}
    />
  </View>
</RCTScrollView>
`;
